target/linux/ramips/dts/mt7620n_hootoo_tm05.dts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Device Tree file for the HooToo TM-05
+ *
+ * Copyright (C) 2020 Russell Morris (rmorris@rkmorris.us)
+ * 
+ * This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+/dts-v1/;
+
+#include "mt7620n.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "hootoo,tm05", "ralink,mt7620n-soc";
+	model = "HooToo TM05";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "tm05:blue:power";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wifi {
+			label = "tm05:green:wifi";
+			gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&i2c {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
```

Comment:
If you want you can also add support for uboot-envtools by adding the config here, and removing `read-only;`

https://github.com/openwrt/openwrt/blob/master/package/boot/uboot-envtools/files/ramips

Reply:
OK, will do - but as a separate PR, right?

Reply:
You may include it here or do it separately. However, this should not be done "for fun", but only if there is a reason to be able to edit the u-boot-env data.

Reply:
Got it. I'll leave this out for now then, can always add later if needed (separate PR).

Reply:
Actually, it hit me - as this is a sealed device, no serial port access (without some surgery, like I did ... LOL), I think this may be good to add in, agreed?

If you concur, I'll go ahead and include it, as we're still fine tuning this.

Reply:
What Adrian, is getting at is. This should be done if you or anyone else would find it necessary to read/write the uboot-env. So the question is "have you ever needed to read/write the uboot-env?"

Reply:
Yep, and a very fair question 😄. I have a times wanted to use nc (netconsole), to see the output, help in debugging. I say that because the serial port is not really available. But to turn on nc, you need to modify the uboot env.

Make sense? And good to include?

Thanks!

Reply:
OK, adding this in - but yell if you disagree (and I'll revert it). Thanks!